url/router-normalizers.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Generates a default type error
+ *
+ * @internal
+ */
+function typeError (expected: string, value: any): never
+{
+    throw new Error(`Can't parse value as ${expected}: '${value}'`);
+}
+
+/**
+ * Normalizes the value as optional string.
+ */
+export function optionalStringNormalizer (value: any): string | null
+{
+    return null != value
+        ? "" + value
+        : null;
+}
+
+/**
+ * Normalizes the value as required string.
+ */
+export function stringNormalizer (value: any): string
+{
+    if (value === undefined)
+    {
+        typeError("string", value);
+    }
+
+    return "" + value;
+}
+
+/**
+ * Normalizes the value as required number.
+ */
+export function numberNormalizer (value?: any): number
+{
+    if (typeof value === "number")
+    {
+        return value;
+    }
+
+    typeError("number", value);
+}
```

Comment:
Any plans to ship an out-of-the-box `boolean` normalizer?

Reply:
Sure, send a PR. 😉 

You just need to take a look what `query-string` serializes a bool to.

Reply:
As I refactored it anyway, I just added them.